gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -797,8 +847,14 @@ impl GatewayLightning for ClnRpcService {
             description,
         } = create_invoice_request.into_inner();
 
-        let payment_hash = sha256::Hash::from_slice(&payment_hash)
-            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+        let payment_hash = if payment_hash.is_empty() {
+            return self
+                .create_invoice_with_known_preimage(amount_msat, expiry_secs.into(), description)
```

Comment:
Naming might be a little weird here: this means "I know the preimage so this is a payment to myself not on behalf of a fedimint client"

Reply:
for_self or something obviosuly "this is mine" might be better

Reply:
Done.